service/matching/tasklist/task_list_manager.go

```diff
@@ -796,31 +797,14 @@ func (c *taskListManagerImpl) getIsolationGroupForTask(ctx context.Context, task
 		}
 		group, err := c.partitioner.GetIsolationGroupByDomainID(ctx, taskInfo.DomainID, partitionConfig, pollerIsolationGroups)
 		if err != nil {
-			// For a sticky tasklist, return StickyUnavailableError to let it be added to the non-sticky tasklist.
-			if err == partition.ErrNoIsolationGroupsAvailable && c.taskListKind == types.TaskListKindSticky {
-				return "", _stickyPollerUnavailableError
-			}
 			// if we're unable to get the isolation group, log the error and fallback to no isolation
 			c.logger.Error("Failed to get isolation group from partition library", tag.WorkflowID(taskInfo.WorkflowID), tag.WorkflowRunID(taskInfo.RunID), tag.TaskID(taskInfo.TaskID), tag.Error(err))
-			return defaultTaskBufferIsolationGroup, nil
+			return defaultTaskBufferIsolationGroup, noIsolationTimeout, nil
```

Comment:
are there any conditions where isolationDuration being returend is ever not -1? sorry, I may have missed it.

Reply:
Not currently, a future PR will add support that for. The goal of this PR is to be little to no change in current behavior because these changes are somewhat hard to put behind a feature flag. Actually having an isolationDuration other than -1 will be behind the feature flag so that we can easily turn it on/off.